infrastructure/infrastructure/src/main/java/alabno/utils/FileUtils.java

```diff
@@ -41,4 +42,52 @@ public static String read_file(String file_url) {
         }
     }
 
+    /**
+     * Split the file on new-line
+     * and group the lines in a list.
+     * N.B.: the lines are indexed from 0 in the list.
+     * @param filePath
+     * @return list of lines
+     */
+    public static List<String> splitFileOnNewLine(Path filePath) {
+
+        List<String> linesList = new ArrayList<>();
+        File file = new File(filePath.toString());
+        try {
+            BufferedReader br = new BufferedReader(new FileReader(file));
+            String line;
+            while ((line = br.readLine()) != null) {
+                linesList.add(line);
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return linesList;
+    }
+
+    /**
+     * Extract file name from path with parent directories
+     * (up to commitX/ exclusive.)
+     * @param filePath
+     * @return file name
+     */
+    public static String getFileName(Path filePath) {
+
+        // Extract final name
+        String fileName = filePath.getFileName().toString();
+
+        // Work backwards to get everything up to .../commitX/
+        Path parentPath = filePath.getParent();
+        Pattern pattern = Pattern.compile("commit.");
```

Comment:
you should put this one outside of the method and make it final, possibly even static